src/neo-vm/Unsafe.cs

```diff
@@ -6,12 +6,12 @@ namespace Neo.VM
     unsafe internal static class Unsafe
     {
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static bool MemoryEquals(byte[] x, byte[] y)
+        public static bool MemoryEquals(ReadOnlySpan<byte> x, ReadOnlySpan<byte> y)
         {
-            if (ReferenceEquals(x, y)) return true;
-            if (x is null || y is null) return false;
+            if (x == y) return true;
```

Comment:
Acording to the source code of `ReadOnlySpan` it doesn't check the null values
https://github.com/dotnet/coreclr/blob/340d69c20d57a2b161aa74c516886bf45df8ee22/src/System.Private.CoreLib/shared/System/ReadOnlySpan.cs#L315

So i recommend to maintain
```if (x is null || y is null) return false;```

Reply:
`ReadOnlySpan` is not a nullable type. It's a `struct`.